launcher/modManager/chroniclesextractor.cpp

```diff
@@ -41,37 +41,23 @@ void ChroniclesExtractor::removeTempDir()
 	tempDir.removeRecursively();
 }
 
-int ChroniclesExtractor::getChronicleNo(QFile & file)
+int ChroniclesExtractor::getChronicleNo()
 {
-	if(!file.open(QIODevice::ReadOnly))
+	for (size_t i = 1; i < chronicles.size(); ++i)
 	{
-		QMessageBox::critical(parent, tr("The file cannot be opened"), file.errorString());
-		return 0;
-	}
+		QString chronicleName = chronicles.at(i);
 
-	QByteArray magic{"MZ"};
-	QByteArray magicFile = file.read(magic.length());
-	if(!magicFile.startsWith(magic))
-	{
-		QMessageBox::critical(parent, tr("Invalid file selected"), tr("You have to select a gog installer file!"));
-		return 0;
-	}
+		QStringList appDirCandidates = tempDir.entryList({"app"}, QDir::Filter::Dirs);
+		QDir appDir = tempDir.filePath(appDirCandidates.front());
```

Comment:
Can the list be empty?

Reply:
Should never happen.
We could try raising some error, in case of some unsupported installer or some other unknown case.

Need to check code though, this was based on some already existing code that also made no checks for emptyness.